release/train_tests/benchmark/image_classification/torch_parquet_image_iterable_dataset.py

```diff
@@ -0,0 +1,493 @@
+from typing import Iterator, List, Optional, Tuple
+import time
+import logging
+import io
+import pandas as pd
+import pyarrow.parquet as pq
+import multiprocessing
+import boto3
+from botocore.exceptions import NoCredentialsError
+
+import torch
+from torch.utils.data import IterableDataset
+
+import ray.train
+import ray
+
+from image_classification.imagenet import get_preprocess_map_fn
+
+logger = logging.getLogger(__name__)
+
+# Set multiprocessing start method to 'spawn' for CUDA compatibility
+if torch.cuda.is_available():
+    try:
+        multiprocessing.set_start_method("spawn", force=True)
+        logger.info(
+            "[DataLoader] Set multiprocessing start method to 'spawn' for CUDA compatibility"
+        )
+    except RuntimeError:
+        logger.info("[DataLoader] Multiprocessing start method already set")
+
+# AWS configuration
+AWS_REGION = "us-west-2"
+
+
+@ray.remote
+def _fetch_parquet_metadata(bucket: str, key: str, file_url: str) -> Tuple[str, int]:
```

Comment:
maybe this can be a 0 CPU task since it's not doing any work (just doing an HTTP request)